bindings/matrix-sdk-crypto-js/src/backup_recovery_key.rs

```diff
@@ -0,0 +1,176 @@
+    #[wasm_bindgen(getter_with_clone)]
+    pub private_key_salt: JsString,
+    /// The number of PBKDF rounds that were used for key derivation.
+    pub private_key_iterations: i32,
+}
+
+/// The public part of the backup key.
+#[derive(Debug, Clone)]
+#[wasm_bindgen]
+pub struct MegolmV1BackupKey {
+    inner: InnerMegolmV1BackupKey,
+    passphrase_info: Option<PassphraseInfo>,
+}
+
+#[wasm_bindgen]
+impl MegolmV1BackupKey {
+    /// The actual base64 encoded public key.
+    #[wasm_bindgen(getter, js_name = "publicKeyBase64")]
+    pub fn public_key(&self) -> JsString {
+        self.inner.to_base64().into()
+    }
+
+    /// The passphrase info, if the key was derived from one.
+    #[wasm_bindgen(getter, js_name = "passphraseInfo")]
+    pub fn passphrase_info(&self) -> Option<PassphraseInfo> {
+        self.passphrase_info.clone()
+    }
+
+    /// Get the full name of the backup algorithm this backup key supports.
+    #[wasm_bindgen(getter, js_name = "algorithm")]
+    pub fn backup_algorithm(&self) -> JsString {
+        self.inner.backup_algorithm().into()
+    }
+
+    /// Signatures that have signed our backup key.
+    /// map of userId to map of deviceOrKeyId to signature
+    #[wasm_bindgen(getter, js_name = "signatures")]
+    pub fn signatures(&self) -> JsValue {
+        let signatures: HashMap<String, HashMap<String, String>> = self
+            .inner
+            .signatures()
+            .into_iter()
+            .map(|(k, v)| (k.to_string(), v.into_iter().map(|(k, v)| (k.to_string(), v)).collect()))
+            .collect();
+
+        serde_wasm_bindgen::to_value(&signatures).unwrap()
+    }
+}
+
+impl BackupRecoveryKey {
+    const KEY_SIZE: usize = 32;
+    const SALT_SIZE: usize = 32;
+    const PBKDF_ROUNDS: i32 = 500_000;
+}
+
+#[wasm_bindgen]
+impl BackupRecoveryKey {
+    /// Create a new random [`BackupRecoveryKey`].
+    #[wasm_bindgen(js_name = "createRandomKey")]
+    pub fn create_random_key() -> BackupRecoveryKey {
+        BackupRecoveryKey {
+            inner: RecoveryKey::new()
+                .expect("Can't gather enough randomness to create a recovery key"),
+            passphrase_info: None,
+        }
+    }
+
+    /// Try to create a [`BackupRecoveryKey`] from a base 64 encoded string.
+    #[wasm_bindgen(js_name = "fromBase64")]
+    pub fn from_base64(key: String) -> Result<BackupRecoveryKey, JsError> {
+        Ok(Self { inner: RecoveryKey::from_base64(&key)?, passphrase_info: None })
+    }
+
+    /// Try to create a [`BackupRecoveryKey`] from a base 58 encoded string.
+    #[wasm_bindgen(js_name = "fromBase58")]
+    pub fn from_base58(key: String) -> Result<BackupRecoveryKey, JsError> {
+        Ok(Self { inner: RecoveryKey::from_base58(&key)?, passphrase_info: None })
+    }
+
+    /// Create a new [`BackupRecoveryKey`] from the given passphrase.
+    #[wasm_bindgen(js_name = "newFromPassphrase")]
+    pub fn new_from_passphrase(passphrase: String) -> BackupRecoveryKey {
```

Comment:
Are we certain that EW needs this? If so perhaps we should give in and put it into the crypto crate instead of having it twice in the bindings.

Reply:
I didn't see it in web code, we can probably reove